extensions/json-language-features/client/src/jsonClient.ts

```diff
@@ -471,6 +470,22 @@ export async function startClient(context: ExtensionContext, newLanguageClient:
 		}
 	}
 
+	async function getSortTextEdits(document: TextDocument, tabSize: string | number = 4, insertSpaces: string | boolean = true): Promise<TextEdit[]> {
+		const filesConfig = workspace.getConfiguration('files', document);
+		const options: SortOptions = {
+			tabSize: Number(tabSize),
+			insertSpaces: Boolean(insertSpaces),
```

Comment:
Previously, this line was:
```
textEditor.options.insertSpaces ? Boolean(textEditor.options.insertSpaces) : true,
```
The previous code was incorrect. The expected inputs are:
* undefined
* false
* true
The previous code would convert them in the following way:
* undefined => true
* false => true
* true => true
So the previous code was incorrect.

The new code does the following:
* undefined => false
* false => false
* true => true

The new code is slightly better, but the defaults of VS Code are that insertSpaces is true if not specified. So I would suggest writing the code like:

```
typeof textEditor.options.insertSpaces === 'undefined' ? true : Boolean(textEditor.options.insertSpaces)
```

Reply:
As I wrote in the comment above, the function takes in the following parameters

```typescript
getSortTextEdits(document: TextDocument, tabSize: string | number = 4, insertSpaces: string | boolean = true):
```

So if insertSpaces is undefined, it should take on the default value `true`.

Reply:
You're right!